float64 x
float64 y
float64 z
float64 ang1
float64 ang2
float64 ang3
